CoreStubs/ArchiveControl/module.php

```diff
@@ -54,9 +80,26 @@ public function DeleteVariableData(int $VariableID, int $StartTime, int $EndTime
         throw new Exception('Not implemented');
     }
 
-    public function GetAggregatedValues(int $VariableID, int $AggregationLevel, int $StartTime, $EndTime, $Limit)
+    public function GetAggregatedValues(int $VariableID, int $AggregationSpan, int $StartTime, int $EndTime, int $Limit)
     {
-        throw new Exception('Not implemented');
+        if ($this->Archive[$VariableID]['AggregatedValues'] == []) {
+            throw new Exception('Aggregated data has to be added through the function AC_StubsAddAggregatedValues()');
+        }
+        if ($Limit > 10000 || $Limit == 0) {
+            $Limit = 10000;
+        }
+        $ArchivedData = array_reverse($this->GetVariableData($VariableID, $AggregationSpan)['AggregatedValues'][$AggregationSpan]);
```

Comment:
Ich würde ['AggregatedValues'][$AggregationSpan] hier nicht setzen. Ich finde es inkonsistent wenn du die den selben Variablennamen wie oben verwendest aber dieser ganz andere Daten enthält. Setzt dies doch in Zeile 93. Da finde ich es wesentlich intuitiver.